server/src/test/java/org/elasticsearch/index/query/PrefixQueryBuilderTests.java

```diff
@@ -166,6 +169,17 @@ public void testRewriteIndexQueryToNotMatchNone() throws Exception {
         }
     }
 
+    public void testMaxLengthExceeded() {
```

Comment:
Instead of this, let's create rest tests similar to [here](https://github.com/elastic/elasticsearch/blob/main/rest-api-spec/src/yamlRestTest/resources/rest-api-spec/test/search/30_limits.yml#L118) or [here](https://github.com/elastic/elasticsearch/blob/main/server/src/internalClusterTest/java/org/elasticsearch/search/simple/SimpleSearchIT.java#L464)